libcodechecker/log/log_parser.py

```diff
@@ -0,0 +1,791 @@
+    # with the ARM Procedure Call Standard.
+    '-mapcs',
+    '-fno-merge-const-bfstores$',
+    '-fno-ipa-sra$',
+    '-mno-thumb-interwork$',
+    # ARM specific option.
+    # Prevent the reordering of
+    # instructions in the function prologue.
+    '-mno-sched-prolog',
+    # This is not unknown but we want to preserve asserts to improve the
+    # quality of analysis.
+    '-DNDEBUG$',
+
+    # --- IGNORED --- #
+    '-MM',
+    '-MP',
+    '-MD',
+    '-MV',
+    '-MMD',
+    '-save-temps',
+    # Clang gives different warnings than GCC. Thus if these flags are kept,
+    # '-Werror', '-pedantic-errors' the analysis with Clang can fail even
+    # if the compilation passes with GCC.
+    '-Werror',
+    '-pedantic-errors',
+    '-g(.+)?$',
+    # Link Time Optimization:
+    '-flto',
+    # MicroBlaze Options:
+    '-mxl',
+    # PowerPC SPE Options:
+    '-mfloat-gprs',
+    '-mabi'
+]
+
+IGNORED_OPTIONS = re.compile('|'.join(IGNORED_OPTIONS))
+
+
+# The compilation flags of which the prefix is any of these regular expressions
+# will not be included in the output Clang command. These flags have further
+# parameters which are also omitted. The number of parameters is indicated in
+# this dictionary.
+IGNORED_PARAM_OPTIONS = {
+    '-MT': 1,
+    '-MQ': 1,
+    '-MF': 1,
+    '-MJ': 1,
+    '-install_name': 1,
+    '-exported_symbols_list': 1,
+    '-current_version': 1,
+    '-compatibility_version': 1,
+    '-init$': 1,
+    '-e$': 1,
+    '-seg1addr': 1,
+    '-bundle_loader': 1,
+    '-multiply_defined': 1,
+    '-sectorder': 3,
+    '--param$': 1,
+    '-u$': 1,
+    '--serialize-diagnostics': 1,
+    '-framework': 1,
+    # Skip paired Xclang options like "-Xclang -mllvm".
+    '-Xclang': 1
+}
+
+
+COMPILE_OPTIONS = [
+    '-nostdinc',
+    r'-nostdinc\+\+',
+    '-pedantic',
+    '-O[1-3]',
+    '-Os',
+    '-std=',
+    '-f',
+    '-m',
+    '-Wno-',
+    '--sysroot=',
+    '--gcc-toolchain='
+]
+
+COMPILE_OPTIONS = re.compile('|'.join(COMPILE_OPTIONS))
+
+
+COMPILE_OPTIONS_MERGED = [
+    '--sysroot',
+    '--include',
+    '-include',
+    '-iquote',
+    '-[DIUF]',
+    '-idirafter',
+    '-isystem',
+    '-macros',
+    '-isysroot',
+    '-iprefix',
+    '-iwithprefix',
+    '-iwithprefixbefore'
+]
+
+COMPILE_OPTIONS_MERGED = \
+    re.compile('(' + '|'.join(COMPILE_OPTIONS_MERGED) + ')')
+
+
+class ImplicitCompilerInfo(object):
+    """
+    This class helps to fetch and set some additional compiler flags which are
+    implicitly added when using GCC.
+    """
+    # TODO: These dicts are mapping compiler to the corresponding information.
+    # It may not be enough to use the compiler as a key, because the implicit
+    # information depends on other data like language or target architecture.
+    compiler_includes = {}
+    compiler_target = {}
+    compiler_standard = {}
+
+    def __init__(self, compiler_info_file=None):
+        self.compiler_info_file = compiler_info_file
```

Comment:
Do we still need the `compiler_info_file`? I see the corresponding option deleted from the command line.

Reply:
The decision here is that we collect the implicit compiler information to this file for debugging reasons for ourselves in the report directory. However, this can't be an input of the analysis like before this commit lacking any reasonable use-case.